app/models/king.rb

```diff
@@ -4,4 +4,48 @@ def valid_move?(x_end, y_end)
     return true unless did_not_move?(x_end, y_end) || too_far
     false
   end
+
+  def castle_check(new_x, color)
+    define_rooks
+    # Need "prevent yourself from moving into check" method to cover the remaining restrictions
+    return true if new_x == 3 && queen_side_castle_check(color)
+    return true if new_x == 7 && king_side_castle_check(color)
+    false
+  end
+
+  def queen_side_castle_check(color)
+    rook = instance_variable_get("@#{color}_queen_rook")
```

Comment:
Whoa! Nice meta-programming. 

How about using this technique to reduce the code duplication and combine queen_side_castle_check and king_side_castle_check into, say: 

```ruby
def castle_check(color, side)
  rook = instance_variable_get("@#{color}_#{side}_rook")
...
```